content/blog/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages.md

```diff
@@ -0,0 +1,263 @@
+
+My preferred way of working with NuGet packages is to first create a prerelease version of a package and use it in my software. If I'm confident that it works I create a release version.
+
+I've created a single pipeline to support this workflow, using the Multi-stage pipelines feature that's currently in preview. The pipeline:
+
+- builds my solution
+- creates a prerelease version of my package
+- creates a release version of my package
+- automatically publishes the prerelease package to a private Azure DevOps Artifacts feed
+- publishes the release version of the package to nuget.org if I give an approval
+
+In this post I'll go into how the pipeline is set up.
+
+First step is to enable the _Multi-stage pipelines_ preview feature:
+
+- Login to your Azure DevOps environment.
+- Click on your avatar in the top right corner.
+- Choose _Preview features_.
+- Enable the _Multi-stage pipelines_ option.  
+![Multi-stage pipelines preview feature](../../static/images/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages/preview-feature-multi-stage-pipelines.png)
+<!-- ![Multi-stage pipelines preview feature](../../../../../images/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages/preview-feature-multi-stage-pipelines.png) -->
+
+Now that the preview feature is enabled we can start. So, create a new pipeline and connect it to one of the sources that supports YAML, like Azure Repositories or GitHub. You can choose any pipeline template, because we're going to clear it and start from scratch. The [Build, test, and deploy .NET Core apps](https://docs.microsoft.com/azure/devops/pipelines/languages/dotnet-core) documentation is a good starting point if you haven't created a pipeline before.
+
+### Create NuGet packages
+
+Clear the YAML editor and add the following YAML. This will build any projects in your git repository using the .NET Core 2.2 SDK.
+
+```yaml
+trigger:
+- master
+
+stages:
+
+- stage: 'Build'
+  variables:
+    buildConfiguration: 'Release'
+
+  jobs:
+  - job:
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    workspace:
+      clean: all
+
+    steps:
+    - task: UseDotNet@2
+      displayName: 'Use .NET Core sdk'
+      inputs:
+        packageType: sdk
+        version: 2.2.x
+        installationPath: $(Agent.ToolsDirectory)/dotnet
+    
+    - task: DotNetCoreCLI@2
+      displayName: "NuGet Restore"
+      inputs:
+        command: restore
+        projects: '**/*.csproj'
+    
+    - task: DotNetCoreCLI@2
+      displayName: "Build Solution"
+      inputs:
+        command: build
+        projects: '**/*.csproj'
+        arguments: '--configuration $(buildConfiguration)'
+```
+
+There are a few things to note. First the pipeline will trigger on a push to master. Next, we specify the `stages` keyword. Indicating that this is a multi-stage pipeline. In the first stage we'll build the solution and create the packages.
```

Comment:
Het lijkt mij nog handig om iets meer uit te leggen wat een multi-stage pipeline precies is. Misschien dat een lezer niet gelijk snapt dat de het release stukje nu door dit vervangen wordt.

Reply:
Ik heb onderstaande aan de 3e paragraaf toegevoegd ipv hier:

This way a separate build & release pipeline is not necessary anymore and you can have your entire pipeline in YAML stored with your code.